opennlp-tools/src/main/java/opennlp/tools/ml/PluggableParameters.java

```diff
@@ -0,0 +1,67 @@
+package opennlp.tools.ml;
+
+import java.util.Map;
+
+public final class PluggableParameters {
+
+	private Map<String, String> parameterMap;
+	private Map<String, String> reportMap;
+	
+	public PluggableParameters(Map<String,String> parameterMap,Map<String,String> reportMap) {
+		this.parameterMap=parameterMap;
+		this.reportMap=reportMap;
+	}
+
+	public String getStringParam(String key, String defaultValue) {
+
+		String valueString = parameterMap.get(key);
+
+		if (valueString == null)
+			valueString = defaultValue;
+
+		if (reportMap != null)
+			reportMap.put(key, valueString);
+
+		return valueString;
```

Comment:
All our code uses two spaces for an indent and not tabs. Could you please reformat that accordingly?

We have that explained here:
https://opennlp.apache.org/code-conventions.html